ccd-definition/CaseEventToComplexTypes/addHearingBookingDetails/addHearingBookingDetailsGatekeeping.json

```diff
@@ -152,8 +175,8 @@
     "CaseFieldID": "hearingDetails",
     "ListElementCode": "judgeAndLegalAdvisor.otherTitle",
     "EventElementLabel": "Title",
-    "FieldDisplayOrder": 10,
-    "FieldShowCondition": "hearingDetails.judgeAndLegalAdvisor.judgeTitle=\"OTHER\"",
+    "FieldDisplayOrder": 12,
+    "FieldShowCondition": "hearingDetails.judgeAndLegalAdvisor.useAllocatedJudge=\"No\" AND hearingDetails.judgeAndLegalAdvisor.judgeTitle=\"OTHER\"",
```

Comment:
This condition is essentially the same as the one already defined in the complextype, so I think it is redundant and can be removed. Same for condition on judgeTitle

Reply:
oh I see you've actually changed the complex type show condition, so yes this would be needed - but why change it?

Reply:
Yep you're right, I had assumed the same thing. But I found when I did not include these additional fieldShowConditions there was inconsistencies with how it behaved. i.e 
- UI would work fine but full name etc would not be persisted in the tabs. 
- Full name input would re-appear if I selected 'no' to use allocatedJudge then selected 'yes'.

I'm not sure wether caseEventToComplexTypes behaves differently for collection items or wether this is a CCD bug but I found the workaround was to repeat show / hide logic.

Reply:
Complex type was changed as data was not persisting in add hearing details event. But this now causes...

- Full name input would re-appear if I selected 'no' to use allocatedJudge then selected 'yes'.

In the other events also :( investigating now.

Reply:
Resolved in latest commit.